nemo_aligner/models/nlp/gpt/megatron_gpt_spin_model.py

```diff
@@ -400,18 +397,95 @@ def onload_adam_states(self):
             self.distributed_adam_offload_manager.__exit__(None, None, None)
 
         self.distributed_adam_offload_manager = None
-
-    def sharded_state_dict(self, prefix: str = ""):
-        sharded_state_dict = super().sharded_state_dict(prefix=prefix)
-
-        # add in the reference policy weights
+    
+    # Alternative 1: swap in the ref weights and create a custom key which then needs to be handled in the loader
+    def sharded_state_dict_alt_1(self, prefix: str = ''):
```

Comment:
Is this working? If yes and you want to keep it, then can you just add a comment to explain that it's not currently used but it's also a valid approach so we keep it around for reference.
(and if not then let's just remove it -- also same thing for `on_load_checkpoint()`)

Reply:
Yes, it works correctly, I will add a comment explaining that it is

Reply:
Added